### DISCUSSION TITLE
Bonsai treadmill input

### DISCUSSION THREAD

**Discussion by skooiman** [bonsai-rx/bonsai#999] [General]
Hello I am a new bonsai user and I am wondering if anyone knows of any resources that would help me with learning how to control a bonvision environment (ex corridor vr)  with a teensy board connected treadmill, controlled by arduino. I am just learning how to work with both arduino and bonsai, but I am assuming that there are workflow examples of how to control something like the z position of an environment such as the vr corridor from bonvision's github. Any help on how to get this treadmill to control the corridor vr would be appreciated, Thank you.

**Comment by bruno-f-cruz** [comment]
Heya,

Indeed, you can start with the VR Corridor demo (https://github.com/bonvision/examples/tree/master/Demos/CorridorVR). In this demo the input to the position is the MouseWheel. However you can connect a different input to this variable. Can you perhaps provide an example on how you are reading the treadmill values in Bonsai? Maybe include a workflow?
Cheers,
B

**Reply by skooiman** [reply]
Hello, thank you for the response. Being new to bonsai, I am using the exact workflow you referenced (vr corridor demo ) to attempt to learn how to manipulate the bonsai inputs. I am currently attempting to figure out how to connect to a different input to the zpos node (specifically a mouse treadmill connected to a teensy board). I was hoping there were workflow examples of how to connect but I have been unable to find any. I have been able to read in the treadmill values two different ways. 1. serial string read and 2. Analog/digital input after loading firmata into the teensy board. However, after getting these inputs I am unsure of how to connect them so they will control the zpos.

Thank you for any help you can offer

**Reply by bruno-f-cruz** [reply]
Replace the mouse input inside `GroupWorkflow` by your Arduino analog input or serial read.

![image](https://user-images.githubusercontent.com/7049351/183384622-01a6e869-5005-4287-8a75-eb1f7511e04c.png)

**Reply by skooiman** [reply]
Ok thank you. When using serial string read it says "No method overload found for the given arguments". I'm assuming if I wanted to use this I would need to find a node that would convert the strings into ints. Using Analog input I am getting readings from the board, just not controlled by the treadmill. I'm assuming this is a problem with the way I integrated my encoder interface code, with the firmata code. I will try to found a way to convert the strings first and if that doesn't work I will try to alter the code on the board. Thank you again for the help.

**Reply by Alphaseltzer** [reply]
Were you able to resolve this?  I would love to know how!

**Reply by bruno-f-cruz** [reply]
Please include the code with your issue. Thanks!
